charts/mimir-distributed/templates/admin-api/admin-api-dep.yaml

```diff
@@ -0,0 +1,120 @@
+{{- if .Values.enterprise.enabled -}}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  annotations:
+    {{- toYaml .Values.admin_api.annotations | nindent 4 }}
+  labels:
+    {{- include "mimir.adminApiLabels" . | nindent 4 }}
+    {{- if .Values.useGEMLabels }}{{- include "mimir.gemExtraLabels" . | nindent 4 }}{{- end }}
```

Comment:
what's the relationship between `Values.enterprise.enabled` and `Values.useGEMLabels`? Can we assume that we always want `Values.useGEMLabels` when in a template guarded by `Values.enterprise.enabled`?

Reply:
I think the two are orthogonal. The `Values.enterprise.enabled` turns on/off licensed features, and `Values.useGEMLabels` is selects between kubernetes standard labels and labels matching GEM 1.8.1 - the later required for the upgrade to work.

`useGEMLabels` is probably not the best name,  something like `enterprise.legacyLabels` would be better. IMHO the current way we generate labels is a bit complicated and has lots of copy paste, not to mention it generates the labels in a different order from GEM 1.8.1, so `helm diff` output is too long. I'm writing a PR to solve these issues.

Reply:
See https://github.com/grafana/helm-charts/pull/1212